Fixed a couple postgres type errors, new type unit tests and insert REAL...

BridgeAR opened the pull request:
... and DOUBLE data types.

mickhansen commented:
Not possible to remove data types, but you can map FLOAT to REAL or DOUBLE.
Actually i thought that FLOAT already mapped to DOUBLE but maybe not.

BridgeAR commented:
This will fix #1157.

mickhansen commented on test/unit/sql/data-types.test.js:
```diff
@@ -161,87 +165,237 @@ suite(Support.getTestDialectTeaser('SQL'), function() {
+      });
     });
 
+    if (current.dialect.name !== 'postgres') {
```
Use `supports` instead.

mickhansen replied:
Actually just kill the IF entirely and provide what we actually do, we do some sort of transformation at the moment.

mickhansen commented on lib/dialects/sqlite/data-types.js:
```diff
@@ -97,11 +97,41 @@ FLOAT.prototype.toSql = function() {
   return NUMBER.prototype.toSql.call(this);
 };
 
+var DOUBLE = function(length, decimals) {
```
I can't tell what's different here for these two types compared to the abstract one.

BridgeAR replied:
The difference is the order of the parameters.

E.g.: 
 'REAL(11,12) UNSIGNED',
sqlite: 'REAL UNSIGNED(11,12)',

mickhansen replied:
Ah yes of course, you are calling the SQLITE type `NUMBER` which handles that.

mickhansen commented on lib/dialects/postgres/data-types.js:
```diff
@@ -52,24 +52,24 @@ DATE.prototype.toSql = function() {
-
-  // POSTGRES does not support length on INTEGER
-  delete this._length;
-  delete this.options.length;
```
Why are you removing this?

BridgeAR replied:
It'll apply the arguments first and remove them afterwards again. PG does not support length, zerofill or unsigned and all possible arguments would have to be removed again.

mickhansen replied:
Ah because you are removing the super call, i'm not sure that's the best way to go, it'd be too easy to add logic to the super and then forget to copy it down here.

mickhansen replied:
I'm aware, hence why i wrote the code to remove it, but it's better to always call the super method when you have inheritance code.

BridgeAR replied:
delete is really slow, so I just assign an empty string instead

mickhansen commented:
What happens if you use FLOAT currently in Postgres?

mickhansen commented on test/unit/sql/data-types.test.js:
```diff
@@ -103,11 +103,13 @@ suite(Support.getTestDialectTeaser('SQL'), function() {
       testsql('INTEGER.UNSIGNED.ZEROFILL', DataTypes.INTEGER.UNSIGNED.ZEROFILL, {
-        default: 'INTEGER UNSIGNED ZEROFILL'
+        default: 'INTEGER UNSIGNED ZEROFILL',
+        postgres: 'INTEGER'
```
Strange that this was passing before.

mickhansen commented:
`CREATE TABLE test (lol FLOAT)` is a valid PG query, PG turns it into a double.

mickhansen commented:
> delete is really slow, so I just assign an empty string instead
That doesn't really matter when the code is run almost never ;p

BridgeAR commented:
This should be ready to merge.

I inserted REAL, DOUBLE and NUMERIC as new data types. NUMERIC is just a synonym for DECIMAL.

Besides that I also fixed a couple of data type errors in Postgres (they do not support zerofill, unsigned and length on a couple of data types). If someone uses any of those in Postgres, they will be stripped and fixed data type will be used instead.
I suggest to warn the user in near future when using data types in a way that is not supported, but this will be in a separate PR.

I also changed FLOAT in Postgres to use DOUBLE instead. Postgres accepts FLOAT with an affinity to DOUBLE but it should be DOUBLE from the beginning on.

And I inserted a couple new unit tests.
